addons/tidb/templates/clusterdefinition.yaml

```diff
@@ -5,138 +5,27 @@ metadata:
-      service:
-        ports:
-          - name: peer
-            port: 20160
-            targetPort: peer
-      podSpec:
-        containers:
-          - name: tikv
-            imagePullPolicy: IfNotPresent
-            ports:
-              - containerPort: 20160
-                name: peer
-            volumeMounts:
-              - name: data
-                mountPath: /var/lib/tikv
-            command:
-              - bin/sh
-              - -c
-              - |
-                echo "start tikv..."
-                DOMAIN=$KB_NAMESPACE".svc{{ .Values.clusterDomain }}"
-                exec /tikv-server --pd=http://${KB_CLUSTER_NAME}-pd.${DOMAIN}:2379 \
-                --data-dir=/var/lib/tikv \
-                --addr=0.0.0.0:20160 \
-                --advertise-addr=${KB_POD_FQDN}:20160 \
-                --status-addr=0.0.0.0:20180
-    - name: tidb
-      characterType: tidb
-      workloadType: Stateful
-      statefulSpec:
-        updateStrategy: BestEffortParallel
-      service:
-        ports:
-          - name: client
-            port: 4000
-            targetPort: client
-          - name: status
-            port: 10080
-            targetPort: status
-      podSpec:
-        containers:
-          - name: tidb
-            imagePullPolicy: IfNotPresent
-            ports:
-              - containerPort: 4000
-                name: client
-              - containerPort: 10080
-                name: status
-            command:
-              - bin/sh
-              - -c
-              - |
-                echo "start tidb..."
-                DOMAIN=$KB_NAMESPACE".svc{{ .Values.clusterDomain }}"
-                exec /tidb-server --store=tikv \
-                --advertise-address=${KB_POD_FQDN}:4000 \
-                --host=0.0.0.0 \
-                --path=${KB_CLUSTER_NAME}-pd.${DOMAIN}:2379 \
-                --log-slow-query=/var/log/tidb/slowlog
-            volumeMounts:
-              - name: slowlog
-                mountPath: /var/log/tidb
-          - name: slowlog
-            imagePullPolicy: IfNotPresent
-            command:
-              - bin/sh
-              - -c
-              - |
-                echo "start slowlog..."
-                touch /var/log/tidb/slowlog
-                tail -n0 -F /var/log/tidb/slowlog;
-            volumeMounts:
-              - name: slowlog
-                mountPath: /var/log/tidb
-        volumes:
-          - name: slowlog
-            emptyDir: {}
+  topologies:
+    - name: cluster
+      default: true
+      components:
+        - name: tidb-pd
+          compDef: tidb-pd-7
+        - name: tikv
+          compDef: tikv-7
+        - name: tidb
+          compDef: tidb-7
+      orders:
+        provision:
+          - tidb-pd
+          - tikv
+          - tidb
+        terminate:
+          - tidb
+          - tikv
+          - tidb-pd
+        # FIXME: due to kb bugs, if update order is specified, update will not succeed
```

Comment:
What is the bug? Are there any corresponding issues?

Reply:
The observation is the first created component's (pd) annotation `kubeblocks.io/generation` does not equal to cluster's generation. So that component's status is stuck in `Creating`. Though I haven't found the root cause.